src/components/App.js

```diff
@@ -1,50 +1,103 @@
 import React from 'react'
+import { withRouter } from 'react-router'
 import { BrowserRouter as Router, Redirect, Route, Switch } from 'react-router-dom'
 import Header from './shared/Header'
 import Navigation from './shared/Navigation/Navigation'
 import Login from './auth/Login.Form'
 import Signup from './auth/Signup.Form'
 import UsersContainer from './users/Container'
 
+import * as auth from '../api/auth'
+
 class App extends React.Component {
-  constructor () {
-    super()
-    this.state = {
-      currentUserId: null
+    constructor() {
+        super()
+        this.state = {
+            currentUserId: null,
+            loading: false
+        }
+
+        this.loginUser = this.loginUser.bind(this)
+        this.signupUser = this.signupUser.bind(this)
+        this.logoutUser = this.logoutUser.bind(this)
     }
 
-    this.loginUser = this.loginUser.bind(this)
-    this.signupUser = this.signupUser.bind(this)
-  }
+    async loginUser(user) {
+        await auth.login(user)
+        const userInfo = await auth.profile()
+        console.log(userInfo.user._id)
+        if (userInfo.status == '200') {
+            this.setState({
+                currentUserId: userInfo.user._id
+            })
+        }
+    }
 
-  loginUser (user) {
-    console.log('Logging In User:', user)
-  }
+    async componentDidMount() {
+        const token = window.localStorage.getItem('journal-app')
 
-  signupUser (user) {
-    console.log('Signing Up User:', user)
-  }
+        if (token) {
+            const profile = await auth.profile()
+            console.log(profile.user._id)
+            this.setState({ 
+              currentUserId: profile.user._id, 
+              loading: false 
+            })
+        }
+    }
 
-  render () {
-    return (
-      <Router>
-        <Header />
-        <Navigation currentUserId={this.state.currentUserId} />
-        <Switch>
-          <Route path='/login' exact component={() => {
-            return <Login onSubmit={this.loginUser} />
-          }} />
-          <Route path='/signup' exact component={() => {
-            return <Signup onSubmit={this.signupUser} />
-          }} />
-
-          <Route path='/users' component={UsersContainer} />
-
-          <Redirect to='/login' />
-        </Switch>
-      </Router>
-    )
+    async signupUser(user) {
+      await auth.signUp(user)
+      // const userInfo = await auth.profile()
+      // why does this not work?
+      // console.log(userInfo)
+      // if (userInfo.status == '200') {
+      //     this.setState({
+      //         currentUserId: userInfo.user._id
+      //     })
+      //   withRouter.history.push('/users')
```

Comment:
withRouter is not used correctly here and the way the router is set up the login and signup routes will automatically redirect once a user id has been defined in state.